bundles/core/org.eclipse.smarthome.core.thing/META-INF/MANIFEST.MF

```diff
@@ -35,6 +35,7 @@ Import-Package:
  org.eclipse.smarthome.config.core.dto,
  org.eclipse.smarthome.config.core.status,
  org.eclipse.smarthome.config.core.validation,
+ org.eclipse.smarthome.config.discovery,

```

Comment:
My comment was about this new dependency: we can not introduce it because it will require the o.e.s.config.dicovery bundle to be present on all solutions built from ESH. It should be possible though to build a solution without a discovery service.

Reply:
But you seemed to have a solution to avoid this dependency ?

Reply:
Unfortunately not. At Leats not thought through to the end.
Here are some points:

* Discovery in bridges could be done in the bridge itself by implementing a new service interface wit a method `provideDiscopveryResults`. For this the class `DiscoveryResult` must be duplicated (not nice)  or moved from config.discovery to core.thing (not possible).
* The bridge could itself report new discovery results to a new method in the ThingHandlerCallback (or better: a new BridgeHandlerCallback) so the ThingManager is then able to provide these new DiscoveryResults to the framework. Again, here its not clear how these results can end up in the config.discovery bundle. One idea would be to issue an event in the event bus. This would however then lead to a public API, which is again not nice.

So only rough ideas. The advantage would be that binding developers would not have to deal with OSGi services at all to provide new discovery results from a bridge.